### PR TITLE
feat(core): add source processor for svg component

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/core/components/svg/svg.component.ts [hunk 1]
```diff
@@ -46,10 +50,13 @@ export class TuiSvgComponent {
     private readonly isIE = isIE(this.userAgent);
     private icon = '';
 
+    @Input()
+    processor: TuiStringHandler<string> = this.globalProcessor;
```
Comment (nsbarsukov) [inline projects/core/components/svg/svg.component.ts:54]: do we need `@Input()` if we already have DI variable ?

I think DI is enough (user can override processor on any level of DI scope).

File: projects/core/tokens/svg-src-processor.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import {InjectionToken} from '@angular/core';
+import {identity, TuiStringHandler} from '@taiga-ui/cdk';
+
+export const TUI_ICON_SRC_PROCESSOR: InjectionToken<TuiStringHandler<string>> =
```
Comment (waterplea) [inline projects/core/tokens/svg-src-processor.ts:4]: TUI_SVG_SRC_PROCESSOR

File: projects/core/components/svg/svg.component.ts (hunk 1 above)
Comment (splincode) [reply projects/core/components/svg/svg.component.ts]: fixed

File: projects/core/tokens/svg-src-processor.ts (hunk 2 above)
Comment (splincode) [reply projects/core/tokens/svg-src-processor.ts]: fixed

File: projects/core/tokens/svg-src-processor.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import {InjectionToken} from '@angular/core';
+import {identity, TuiStringHandler} from '@taiga-ui/cdk';
+
+export const TUI_SVG_SRC_PROCESSOR: InjectionToken<TuiStringHandler<string>> =
+    new InjectionToken<TuiStringHandler<string>>('Source path processor for svg', {
+        factory: () => identity,
```
Comment (waterplea) [inline projects/core/tokens/svg-src-processor.ts:6]: Let's use `identity` from rxjs and deprecate our `identity`.

**Review by waterplea — APPROVED** [review]

File: projects/core/tokens/svg-src-processor.ts (hunk 3 above)
Comment (splincode) [reply projects/core/tokens/svg-src-processor.ts:7]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**